### PR TITLE
app: add tauri github action to build app for mac and linux

### PR DESCRIPTION
**Pull request by burmudar**
Add an action that runs when commits are pushed to the app release branch
## Test plan
Tested action on this branch - successful run https://github.com/sourcegraph/sourcegraph/actions/runs/4798442912

### CODE REVIEW

**Review by chwarwick — APPROVED** [review]

File: src-tauri/tauri.conf.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
   },
   "package": {
     "productName": "Sourcegraph App",
-    "version": "1.0.0-dev"
+    "version": "1.0.0"
```
Comment (chwarwick) [inline src-tauri/tauri.conf.json]: @burmudar  Should the version here align with the "backend" version? I think this is what Tauri uses when checking for updates?  I'm almost positive we also need to remove this [hack](https://sourcegraph.sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/enterprise/cmd/sourcegraph/BUILD.bazel?L10:73)

Comment (burmudar) [reply src-tauri/tauri.conf.json]: @chwarwick So currently the frontend and the backend versions are specified differently. The backend gets compiled normally with a backend friendly version number like `5.0.3`. Then the frontend gets compiled with the frontend app version number which is `2023.5.8+{buildNumber}.{SHA}`. You can see the latest `build-release.sh` for how it's done

File: enterprise/dev/app/build-release.sh [hunk 2]
```diff
@@ -5,52 +5,79 @@ set -eu
...
+    if [[ ${RELEASE_BUILD} == 1 ]]; then
+      build="release"
+    fi
+    echo "$(date '+%Y.%-m.%-d')+${build}.${sha}"
+}
 
+set_version() {
+  if [[ ${CI:-""} == "true" ]]; then
+    # Backend version
+    export VERSION=$(cat ./src-tauri/VERSION)
```
[the remark this answers is not in the document]
Comment (burmudar) [reply enterprise/dev/app/build-release.sh:36]: I'll add a comment in `app_stamp_vars.sh` to make the assumption clear that `build-release.sh` is assumed to have been executed

File: src-tauri/tauri.conf.json (hunk 1 above)
Comment (burmudar) [reply src-tauri/tauri.conf.json]: Yup. That version will be `5.0.3`, we can bake extra information into that version string - I have to look at what OOB migrations uses from that string like `5.0.3+app.{SHA}` etc.

**Pull request merged**